scripts/track-dependency-status.sh

```diff
@@ -0,0 +1,48 @@
+# Get total count of python dependencies and outdated python dependencies
+
+total_python_deps="$(pip list| wc -l)"
+outdated_python_deps="$(pip list --outdated| wc -l)"
+
+# Get outdated JS dependency count
+
+COUNT_OUTDATED_DEP_CODE=$(cat <<-END
+import json
+import sys
+
+dep_arr = json.load(sys.stdin)
+print(len(dep_arr["data"]["body"]))
+
+END
+)
+
+outdated_js_deps="$(yarn outdated --json | sed -n 2p | python -c "$COUNT_OUTDATED_DEP_CODE")"
+
+
+# Get total JS dependencies from package.json
+COUNT_DEP_CODE=$(cat <<-END
+import json
+import sys
+
+file = open('package.json', 'r')
+file_obj=json.load(file)
+
+deps= len(file_obj["dependencies"].keys())
+dev_deps = len(file_obj["devDependencies"].keys())
+print(deps+dev_deps)
+
+file.close()
+END
+)
+total_js_deps="$(python -c "$COUNT_DEP_CODE")"
```

Comment:
Following the theme of the above suggestion, this could be

```suggestion
function dependency_count {
    local json_path="$1"
    python - "$json_path" <<-END
import json
import sys

with open(sys.argv[1], "r") as json_file:
    packages = json.load(json_file)

print(sum(len(packages[k]) for k in ["dependencies", "devDependencies"]))
END
}

total_js_deps=$(dependency_count package.json)
```

Reply:
Is the `jq` command available? If so, this could alternatively be reduced to:

```bash
total_js_deps=$(jq -r '(.dependencies + .devDependencies)|keys[]' package.json | wc -l)
```